custom_components/ev_smart_charging/helpers/config_flow.py

```diff
@@ -185,6 +188,21 @@ def find_generic_sensor(hass: HomeAssistant) -> str:
                     return entity_id
         return ""
 
+    @staticmethod
+    def find_tge_sensor(hass: HomeAssistant) -> str:
+        """Search for TGE sensor"""
+        entity_registry: EntityRegistry = async_entity_registry_get(hass)
+        registry_entries: UserDict[str, RegistryEntry] = (
+            entity_registry.entities.items()
+        )
+        for entry in registry_entries:
+            if entry[1].platform == PLATFORM_TGE:
+                entity_id = entry[1].entity_id
+                attributes = hass.states.get(entity_id).attributes
+                if all(attr in attributes for attr in ["prices", "prices_today", "prices_tomorrow"]):
```

Comment:
It would be more effective and more reliable to search entities by unique id. In case of Fixing 1 it would be `tge_sensor_fixing1_rate`

Reply:
Thank you! I commited this improvement.